ir/shared/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -1258,7 +1263,27 @@ object Serializers {
 
         case TagJSNew                => JSNew(readTree(), readTreeOrJSSpreads())
         case TagJSPrivateSelect      => JSPrivateSelect(readTree(), readFieldIdent())
-        case TagJSSelect             => JSSelect(readTree(), readTree())
+        case TagJSSelect             =>
+          readTree() match {
+            case qualifier: JSLinkingInfo if hacks.use16 =>
```

Comment:
This won't work, because `readTree()` will already apply the hack that turns `JSLinkingInfo()` into a `JSObjectConstr`.

I think we can hack it this way:
```scala
if (hacks.use16 && buf.getByte(buf.position()) == TagJSLinkingInfo) {
  val jsLinkingInfo = readTree()
  readTree() match {
    case StringLiteral("productionMode") =>
      ...
    case otherItem =>
      JSSelect(jsLinkingInfo, otherItem)
} else {
  JSSelect(readTree(), readTree())
}
```
Note that we must read the `jsLinkingInfo` tree to consume the tag and position. Moreover, for an unknown property, we must not emit an error. We must emit the default `JSSelect` over the expansion `jsLinkingInfo`. It is valid IR that will evaluate to `undefined`.